### PR TITLE
NBL-17: Create product details screen

### PR DESCRIPTION
**Pull request by abellnoel**
- Create product details component
- Add ID to product interface
- Add getProduct to product service retrieving a single product by ID
- Creating routing for `.../product/id`
- Route "View Details" buttons to product details page passing in their ID

### CODE REVIEW

File: src/components/product-card/product-card.component.html [hunk 1]
```diff
@@ -1,46 +1,50 @@
...
-<img src="{{product.imageURL}}"
-    class="card-img-top" alt="spongebob chair" />
-<div class="card-body">
-    <div class="text-center">
-    <h5 class="card-title">{{product.productName}}</h5>
-    <p class="text-muted mb-4">{{product.productDetails}}</p>
+    <img
+        src="{{ product.imageURL }}"
+        class="card-img-top"
+        alt="spongebob chair"
```
Comment (twigmytwig) [inline src/components/product-card/product-card.component.html]: I think we can change the alt text to product.productName now

File: src/components/product-card/product-card.component.html [hunk 2]
```diff
@@ -1,46 +1,50 @@
...
+            <h5 class="card-title">{{ product.productName }}</h5>
+            <p class="text-muted mb-4">{{ product.productDetails }}</p>
+        </div>
+        <div>
+            <div class="d-flex justify-content-between">
+                <span>Cost</span
+                ><span>{{ product.productPrice | currency }}</span>
+            </div>
+            <div class="d-flex justify-content-between">
+                <span>Shipping</span
```
Comment (twigmytwig) [inline src/components/product-card/product-card.component.html:18]: I wonder if we should take out the shipping and additional fees part of the product-card/product interface. It feels like we won't be able to calculate shipping and fees until you look at an entire cart.

File: src/components/product-detail/product-detail.component.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            <div class="card h-100">
+                <div class="card-header fs-1">
+                    <h1>{{ product.productName }}</h1>
+                </div>
+                <div class="card-body">
+                    <h2 class="card-title">
+                        {{ product.productPrice | currency }}
+                    </h2>
+                    <p class="card-text">
+                        Shipping: {{ product.shippingPrice | currency }}
```
Comment (twigmytwig) [inline src/components/product-detail/product-detail.component.html:16]: Same inquiry as before. I just wonder if we need a shipping/additional fees ON a product. Maybe we replace it with something like quantityInStock, weight, dimensions, so on so forth. We could just keep it like this for now though its not a huge deal.

File: src/components/product-detail/product-detail.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+@Component({
+    selector: 'nbl-product-detail',
+    standalone: true,
+    templateUrl: './product-detail.component.html',
+    styleUrls: ['./product-detail.component.css'],
+    imports: [CommonModule, TitleComponent],
+})
+export class ProductDetailComponent implements OnInit {
+    product?: Product;
```
Comment (twigmytwig) [inline src/components/product-detail/product-detail.component.ts:17]: This file is very cool i learned some stuff reading this

File: src/interfaces/productInterface.ts [hunk 5]
```diff
@@ -1,10 +1,9 @@
-export interface Product{
+export interface Product {
+    id: string;
     imageURL: string;
     productName: string;
     productDetails?: string; //optional property
     productPrice: number;
     shippingPrice: number;
```
Comment (twigmytwig) [inline src/interfaces/productInterface.ts:7]: Same comment as before again. Do we need this in this interface?

File: src/services/product.service.ts [hunk 6]
```diff
@@ -1,39 +1,46 @@
...
+            feesPrice: 4.0,
+        },
+    ];
 
-        return of(products);
+    getProducts(): Observable<Product[]> {
+        return of(this.mockProducts);
+    }
+
+    getProduct(id: string): Observable<Product | undefined> {
```
Comment (twigmytwig) [inline src/services/product.service.ts:43]: Shouldn't id be of type number? Our ID's will likely only be 1,2,3,4...

File: src/services/shopping-cart.service.ts [hunk 7]
```diff
@@ -1,21 +1,22 @@
...
+    providedIn: 'root',
 })
 export class ShoppingCartService {
+    products: Product[] = [];
 
-  products: Product[] = [];
-
-   addProductToCart(product: Product){
+    addProductToCart(product: Product): void {
         alert('You have added ' + product.productName + ' To your cart!');
```
Comment (twigmytwig) [inline src/services/shopping-cart.service.ts]: I think we can get rid of these alerts now, the cart number incrementing is sufficient for showing this is working

File: src/components/product-card/product-card.component.html (hunk 2 above)
Comment (abellnoel) [reply src/components/product-card/product-card.component.html:18]: Yeah that sounds like a good idea.

Comment (abellnoel) [reply src/components/product-card/product-card.component.html:18]: I'll leave that for another ticket though, would have to change a lot of things.

File: src/services/product.service.ts (hunk 6 above)
Comment (abellnoel) [reply src/services/product.service.ts:43]: From what I've read, I think IDs are better off as strings if we don't plan to do any calculations with them. If we for some reason wanted to add other characters into IDs in the future, we could if we start off with strings.

File: src/services/shopping-cart.service.ts (hunk 7 above)
Comment (abellnoel) [reply src/services/shopping-cart.service.ts]: ~~I'm gonna try adding a toast~~ nvm that'll probably be moreinvolved, I'll jusut remove them for now and create a ticket for the toast.

**Review by twigmytwig — APPROVED** [review]

**Pull request merged**